src/protagonist/API/Features/Image/ImageController.cs

```diff
@@ -83,19 +83,33 @@ public async Task<IActionResult> GetImage(int customerId, int spaceId, string im
     /// <param name="pageSize"></param>
     /// <param name="orderBy"></param>
     /// <param name="orderByDescending"></param>
+    /// <param name="q">A serialised JSON <see cref="ImageQuery"/> object</param>
     /// <returns>A Hydra Collection of Image objects as JSON-LD</returns>
     [HttpGet]
     [ProducesResponseType(200, Type = typeof(HydraCollection<DLCS.HydraModel.Image>))]
     [ProducesResponseType(404, Type = typeof(Error))]
     public async Task<IActionResult> GetImages(
         int customerId, int spaceId,
         int? page = 1, int? pageSize = -1,
-        string? orderBy = null, string? orderByDescending = null)
+        string? orderBy = null, string? orderByDescending = null,
+        string? q = null)
     {
         if (pageSize is null or < 0) pageSize = Settings.PageSize;
         if (page is null or < 0) page = 1;
+        AssetFilter? assetFilter = null;
+        if (q.HasText())
+        {
+            assetFilter = Request.GetAssetFilterFromQParam(q);
+            if (assetFilter == null)
+            {
+                return HydraProblem("Could not parse query", null, 400);
+            }
+        }
+        // Now augment this assetFilter, or create one if we don't have one yet
+        assetFilter = Request.UpdateAssetFilterFromQueryStringParams(assetFilter);

```

Comment:
would lines 99-109 be cleaner as a method call as there's already a few things going on in that method. making it easier when scanning. `AssetFilter? assetFilter = GetAssetFilter();` (personal opinion!)

Reply:
Have rewritten to improve